parquet/src/test/java/org/apache/iceberg/parquet/TestParquetSchemaUtil.java

```diff
@@ -224,7 +237,16 @@ public void testSchemaConversionWithoutAssigningIds() {
                 "map_col_5",
                 Repetition.REQUIRED,
                 primitive(28, "k", PrimitiveTypeName.INT32, Repetition.REQUIRED),
-                primitive(29, "v", PrimitiveTypeName.INT32, Repetition.REQUIRED)));
+                primitive(29, "v", PrimitiveTypeName.INT32, Repetition.REQUIRED)),
+            variant(30, "variant_col_1", Repetition.OPTIONAL),
+            variant(null, "variant_col_2", Repetition.REQUIRED),
+            list(31, "list_col_6", Repetition.OPTIONAL, variant(32, "v", Repetition.OPTIONAL)),
```

Comment:
Can we also add a test with a variant in map?

Reply:
Updated this test, but reached the 150-line Checkstyle limit, so I moved the variant-related tests into separate methods.